main.py

```diff
@@ -24,4 +24,12 @@ def redraw(self, event=None):
                 x2 = x1 + cellwidth
                 y2 = y1 + cellheight
                 tile.gui_tile = self.canvas.create_rectangle(x1,y1,x2,y2, fill=tile.color, tags="rect")
-                
+                if tile.isOccupied():
+                    if tile.piece.isKing:
+                        oval = self.canvas.create_oval(x1,y1,x2,y2, fill=tile.piece.color, width = 10)
+                        self.canvas.tag_bind(oval, "<1>", lambda event, row=row, column=column: self.clicked(row, column))
+                    else:
+                        oval = self.canvas.create_oval(x1,y1,x2,y2, fill=tile.piece.color)
+                        self.canvas.tag_bind(oval, "<1>", lambda event, row=row, column=column: self.clicked(row, column))
+                self.canvas.tag_bind(tile.gui_tile, "<1>", lambda event, row=row, column=column: self.clicked(row, column))
+            
```

Comment:
Lines 8, 9 and 14 aren't being used at all. Delete them, don't be shy 🙆‍♂️

Reply:
really if i'm using it in the width of the widget line 8 and 9

Reply:
Where are you using them?

Reply:
18

Reply:
Oh, all right, so either `cellwidth` or `cellheight` should depend on `self.rows`. Both dimensions are using `self.columns` right now. Or you can use a single variable called `self.dimensions` instead of rows and columns

Reply:
Now that I notice I was not using rows but I will actually use Dimensions

Reply:
It was my mistake I thought that if I was using it xD, but in the same way it is much better to use dimensions instead